src/MonteCarlo/MonteCarlos.jl

```diff
@@ -73,16 +73,26 @@ Expression tree optimization using Monte Carlo with parameters p, grammar 'gramm
 function monte_carlo(p::MonteCarlo, grammar::Grammar, typ::Symbol, loss::Function; 
     verbose::Bool=false)
     dmap = mindepth_map(grammar)
-    best_tree, best_loss = RuleNode(0), Inf
-    for i = 1:p.num_samples
-        verbose && println("samples: $i of $(p.num_samples)")
+
+    trees = Vector{RuleNode}(undef, p.num_samples)
```

Comment:
I could have only allocate a vector of length `Threads.nthreads()`.
It would be a problem for this loop but is an issue for the second loop for the `_update_tracker!` function, which would need to loops over all individual tree and loss.

Reply:
Yep, I agree with the changes that you've made.  It looks good to me.  Thanks for contributing!  I'll merge it.